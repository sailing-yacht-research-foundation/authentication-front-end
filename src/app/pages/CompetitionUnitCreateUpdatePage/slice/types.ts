/* --- STATE --- */
export interface competitionUnitManagerState {
    lastSubscribedCompetitionUnitId: string;
}

export type ContainerState = competitionUnitManagerState;
