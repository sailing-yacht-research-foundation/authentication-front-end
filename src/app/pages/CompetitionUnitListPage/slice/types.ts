/* --- STATE --- */
export interface MyRaceListState {
    results: any[];
    page: number;
    total: number;
    is_changing_page: boolean;
  }
  
  export type ContainerState = MyRaceListState
  