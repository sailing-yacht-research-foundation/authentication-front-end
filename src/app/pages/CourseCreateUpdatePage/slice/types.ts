/* --- STATE --- */
export interface CourseState {
    course_sequenced_geometries: any[];
    page: number;
    total_course: number;
    courses: any[];
    is_handling_course: boolean;
}

export type ContainerState = CourseState;
