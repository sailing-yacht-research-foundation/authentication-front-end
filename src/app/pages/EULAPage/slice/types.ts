/* --- STATE --- */
export interface EULAState { }

export type ContainerState = EULAState
