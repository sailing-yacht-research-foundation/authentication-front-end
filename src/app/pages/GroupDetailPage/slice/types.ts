import { Group, GroupMember } from "types/Group";

/* --- STATE --- */
export interface GroupDetailState {
    adminCurrentPage: number;
    adminTotal: number;
    memberCurrentPage: number;
    memberTotal: number;
    admins: GroupMember[];
    members: GroupMember[];
    isGettingAdmins: boolean;
    isGettingMembers: boolean;
    group: Partial<Group>;
    isGettingGroup: boolean;
    getGroupDetailFailed: boolean;
    acceptedMemberResults: any[];
    adminPageSize: number;
    memberPageSize: number;
  }
  
  export type ContainerState = GroupDetailState;
  