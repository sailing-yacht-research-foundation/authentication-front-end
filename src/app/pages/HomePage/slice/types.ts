/* --- STATE --- */
export interface HomeState {
    results: any[];
    keyword: string;
    from_date: string;
    to_date: string;
    page: number;
    is_searching: boolean;
    total: number;
    page_size: number;
    show_advanced_search: boolean;
    sort: string;
  }
  
  export type ContainerState = HomeState
  