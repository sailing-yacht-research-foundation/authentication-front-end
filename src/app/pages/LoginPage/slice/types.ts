/* --- STATE --- */
export interface LoginState {
  user: any,
  is_authenticated: boolean,
  session_token: string,
  syrf_authenticated: boolean;
  user_coordinate?: null | Coordinate;
}

export type ContainerState = LoginState

export interface Coordinate {
  lat: number;
  lon: number;
}