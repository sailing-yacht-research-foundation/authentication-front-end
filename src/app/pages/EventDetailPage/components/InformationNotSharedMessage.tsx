import React from 'react';
import { Alert, Button } from 'antd';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import { translations } from 'locales/translations';
import { CalendarEvent } from 'types/CalendarEvent';
import { requiredCompetitorsInformation } from 'utils/constants';
import { renderRequirementBasedOnEventKey } from 'utils/helpers';
import { ConfirmSharingInformationModal } from './ConfirmSharingInformationModal';

export const InformationNotShared = ({ event, reloadParent }: { event: Partial<CalendarEvent>, reloadParent: Function }) => {

    const { t } = useTranslation();

    const [isShowSharingInformationConfirmModal, setIsShowSharingInformationConfirmModal] = React.useState<boolean>(false);

    const requiredInformation: any = [];

    Object.keys(event).forEach(key => {
        if (requiredCompetitorsInformation.includes(key) && event[key] === true)
            requiredInformation.push(<li>{renderRequirementBasedOnEventKey(t, key)}</li>);
    });

    if (event.isParticipant
        && !event.participantDetail?.allowShareInformation
        && requiredInformation.length > 0)
        return (
            <Wrapper>
                <ConfirmSharingInformationModal
                    setShowModal={setIsShowSharingInformationConfirmModal}
                    showModal={isShowSharingInformationConfirmModal}
                    event={event}
                    reloadParent={reloadParent}
                    requiredInformation={requiredInformation} />
                <Alert
                    message={t(translations.event_detail_page.this_event_requires_some_of_your_profile_information_click_share_now_if_you_want_to_share_them_with_the_organizer)}
                    type="warning"
                    closable
                    action={
                        <Button type='link' onClick={() => setIsShowSharingInformationConfirmModal(true)}>{t(translations.event_detail_page.share_now)}</Button>
                    }
                />
            </Wrapper>
        );

    return (<></>);
}

const Wrapper = styled.div`
    margin: 10px 0;
    margin-top: 15px;
`;