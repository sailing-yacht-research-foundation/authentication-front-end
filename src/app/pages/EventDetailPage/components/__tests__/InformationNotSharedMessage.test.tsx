import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import Provider from 'app/components/Provider/index';
import { InformationNotShared } from '../InformationNotSharedMessage';
import { fireEvent, render } from '@testing-library/react';
import { CalendarEvent } from 'types/CalendarEvent';
import { i18n } from 'locales/i18n';
import { translations } from 'locales/translations';
import { act } from 'react-test-renderer';

const shallowRenderer = createRenderer();

describe('InformationNotSharedMessage', () => {
    it('should render to match the snapshot', () => {
        shallowRenderer.render(
            <Provider>
                <InformationNotShared event={{}} reloadParent={jest.fn} />
            </Provider>);
        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });

    it('should render the message if the conditional state matches', async () => {
        const event: Partial<CalendarEvent> = {
            participantDetail: {
                allowShareInformation: false,
            },
            isParticipant: true,
            requireEmergencyContact: true,
            requireCovidCertificate: true,
            requireMedicalProblems: true,
            requireFoodAllergies: true,
            requireImmigrationInfo: true
        }
        const { getByText } = render(<Provider>
            <InformationNotShared event={event} reloadParent={jest.fn} />
        </Provider>);
        const t = await i18n;

        expect(getByText(String(t(translations.event_detail_page.this_event_requires_some_of_your_profile_information_click_share_now_if_you_want_to_share_them_with_the_organizer)))).toBeTruthy();
    });

    it('should render nothing if the user allowed to share information', async () => {
        const event: Partial<CalendarEvent> = {
            participantDetail: {
                allowShareInformation: true,
            },
            isParticipant: true,
            requireEmergencyContact: true,
            requireCovidCertificate: true,
            requireMedicalProblems: true,
            requireFoodAllergies: true,
            requireImmigrationInfo: true
        }
        const { container } = render(<Provider>
            <InformationNotShared event={event} reloadParent={jest.fn} />
        </Provider>);

        expect(container.firstChild).toBeNull();
    });

    it('should show ConfirmSharingInformationModal if the the user clicks share now', async () => {
        const event: Partial<CalendarEvent> = {
            participantDetail: {
                allowShareInformation: false,
            },
            isParticipant: true,
            requireEmergencyContact: true,
            requireCovidCertificate: true,
            requireMedicalProblems: true,
            requireFoodAllergies: true,
            requireImmigrationInfo: true
        }

        const { getByText } = render(<Provider>
            <InformationNotShared event={event} reloadParent={jest.fn} />
        </Provider>);
        const t = await i18n;

        act(() => {
            fireEvent.click(getByText(String(t(translations.event_detail_page.share_now))))
        });

        expect(getByText(String(t(translations.event_detail_page.share_information_to_the_organizers)))).toBeInTheDocument();
    });
});
