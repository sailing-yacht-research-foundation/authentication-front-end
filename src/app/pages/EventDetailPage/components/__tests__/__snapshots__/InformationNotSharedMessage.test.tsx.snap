// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationNotSharedMessage should render the message if the conditional state matches 1`] = `
<Provider
  store={
    Object {
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <InformationNotShared
    event={
      Object {
        "participantDetail": Object {
          "allowShareInformation": true,
          "isParticipant": true,
        },
        "requireCovidCertificate": true,
        "requireEmergencyContact": true,
        "requireFoodAllergies": true,
        "requireImmigrationInfo": true,
        "requireMedicalProblems": true,
      }
    }
    reloadParent={[Function]}
  />
</Provider>
`;

exports[`InformationNotSharedMessage should render to match 1`] = `
<Provider
  store={
    Object {
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <InformationNotShared
    event={Object {}}
    reloadParent={[Function]}
  />
</Provider>
`;

exports[`InformationNotSharedMessage should render to match the snapshot 1`] = `
<Provider
  store={
    Object {
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <InformationNotShared
    event={Object {}}
    reloadParent={[Function]}
  />
</Provider>
`;
